### PR TITLE
Test knight

### PR DESCRIPTION
**Pull request by TanTheMan768**
* Add script tag to test.html
* Add test for constructor
* Add test for movement in center of empty board
* Add test for corners of empty board
* Add test for friendly pieces
* Add test for enemy pieces

### CODE REVIEW

File: test/knight-test.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            [4,1], [4, 5], [5,2], [5,4]
+        ];
+        possibleMoves.sort();
+        expectedMoves.sort();
+        assert.deepEqual(
+            JSON.stringify(possibleMoves),
+            JSON.stringify(expectedMoves),
+            "Verify {possibleMoves()} in center of empty board."
+        );
+    })
```
Comment (jbeBan) [inline test/knight-test.js]: Semicolon missing on end of line `68`.

File: test/knight-test.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    /**
+     * Test possibleMoves() with knight at [0,0]
+     */
+    QUnit.test("possibleMoves() at {0,0}", assert => {
+        let knight = new Knight(6969, 0, 0, 0)
```
Comment (jbeBan) [inline test/knight-test.js]: Semicolon missing on end of line `74`.

File: test/knight-test.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            JSON.stringify(expectedMoves),
+            "Verify {possibleMoves() at [0,0]"
+        );
+    });
+
+    /**
+    * Test possibleMoves() with knight at [7,7]
+    */
+    QUnit.test("possibleMoves() at {7,7}", assert => {
+        let knight = new Knight(6969, 7, 7, 0)
```
Comment (jbeBan) [inline test/knight-test.js]: Semicolon missing on end of line `90`.

File: test/knight-test.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            "Verify {column} is 0."
+        );
+        assert.deepEqual(
+            knight.playerId,
+            0,
+            "Verify {playerId} is 0."
+        );
+        assert.deepEqual(
+            knight.unicodeChar,
+            "U+265E",
```
Comment (jbeBan) [inline test/knight-test.js]: Unicode character on line `46` should be `&#9822;`.

**Review by jbeBan — CHANGES_REQUESTED** [review]

File: test/knight-test.js (hunk 1 above)
Comment (TanTheMan768) [reply test/knight-test.js]: Updated.

File: test/knight-test.js (hunk 2 above)
Comment (TanTheMan768) [reply test/knight-test.js]: Updated.

File: test/knight-test.js (hunk 3 above)
Comment (TanTheMan768) [reply test/knight-test.js]: Updated.

File: test/knight-test.js (hunk 4 above)
Comment (TanTheMan768) [reply test/knight-test.js]: Updated.

**Review by jbeBan — APPROVED** [review]
Perfect!
